src/main/java/space/obminyashka/items_exchange/controller/SubcategoryController.java

```diff
@@ -57,7 +59,7 @@ public ResponseEntity<List<String>> getSubcategoryNamesByCategoryId(@PathVariabl
     @ResponseStatus(HttpStatus.OK)
     public void deleteSubcategoryById(@PathVariable("subcategory_id") @Positive(message = "{" + INVALID_NOT_POSITIVE_ID + "}") long id)
             throws InvalidDtoException {
-        if (!subcategoryService.isSubcategoryDeletable(id)) {
+        if (advertisementService.isAdvertisementsHaveSubcategoryWithId(id)) {
```

Comment:
It would be better to rename it to "areAdvertisementsExistWithSubcategory(subcategoryId)"